### PR TITLE
P2p validation on main

### PR DESCRIPTION
**Pull request by LeoHChen**

### CODE REVIEW

**Comment by LeoHChen** [comment]
image

image

The CPU usage on a validator will be doubled as we do pubkey checking on every p2p messages even though validators may ignore them.  This is the tradeoff to reduce the propagation of the invalid messages into the network. We can see clearly that the outgoing network traffic is lower than incoming traffic now as invalid ones are filtered out.

File: node/node_handler.go [hunk 1]
```diff
@@ -56,101 +55,75 @@ func (node *Node) processSkippedMsgTypeByteValue(
...
-		} else {
-			node.ConsensusMessageHandler(msgPayload)
+// HandleNodeMessage parses the message and dispatch the actions.
+func (node *Node) HandleNodeMessage(
+	ctx context.Context,
+	payload []byte,
+) error {
+
+	// Prevent OOB crash
+	if len(payload) < (proto.MessageTypeBytes + proto.MessageTypeBytes) {
```
Comment (JackyWYX) [inline node/node_handler.go:70]: Should it be MessageTypeBytes + MessageCatagoryBytes?

Comment (LeoHChen) [reply node/node_handler.go:70]: fixed in new PR https://github.com/harmony-one/harmony/pull/3212

File: consensus/checks.go [hunk 2]
```diff
@@ -1,99 +1,73 @@
...
-			"Failed to verify sender's signature",
-		)
-		return false
+	msgHash := hash.Keccak256(messageBytes)
+	if !msgSig.VerifyHash(signerPubKey, msgHash[:]) {
+		return errors.New("failed to verify the signature")
 	}
-
-	return true
+	message.Signature = signature
```
[the remark this answers is not in the document]
Comment (LeoHChen) [reply consensus/checks.go:51]: Not sure. that's the original code.

**Comment by LeoHChen** [comment]
New PR rebased based on latest main branch. https://github.com/harmony-one/harmony/pull/3212

**Pull request closed without merging**